### PR TITLE
Update the build and install

### PR DESCRIPTION
**Pull request by mellis13**
In this PR, the build and install process of SmartRedis is improved.  Now, when ``make lib`` is executed by the user, the SmartRedis library and all dependencies will be installed in ``./install``.  This enables users to reference only the ``install`` directory in application build files without the need for environment variables to resolve SmartRedis dependencies.  Examples, tests, and the pip install have been updated to use this change.

### CODE REVIEW

**Review by Spartee — CHANGES_REQUESTED** [review]
- update ``pytest`` to 6.0.0 or greater and change in ``setup.cfg`` under the dev line ``requirements-dev.txt`` (this is because we wil need to use importlib to run tests against installed copies of SmartRedisPy
- add cmake and reference it in all build scripts like we do in ``build_deps.sh``
- Update RedisAI and tests deps with build scripts from SmartSim (optional but recommended)

After these, I should be able to get the tests running in Github Actions

File: setup.py [hunk 1]
```diff
@@ -36,9 +36,9 @@ def run(self):
         check_prereq("gcc")
         check_prereq("g++")
```
[the remark this answers is not in the document]
Comment (mellis13) [reply setup.py:37]: Good catch.   Done.

File: setup_test_env.sh [hunk 2]
```diff
@@ -11,3 +11,21 @@ source ./setup_env.sh
...
+# Redis
+if [[ -f ./third-party/redis/src/redis-server ]]; then
+    export REDIS_INSTALL_PATH="$(pwd)/third-party/redis/src"
+    echo "Set Redis server install path to $REDIS_INSTALL_PATH"
+fi
+
+# detect RedisAI CPU installation
+if [[ -f ./third-party/RedisAI/install-cpu/redisai.so ]]; then
+    export REDISAI_CPU_INSTALL_PATH="$(pwd)/third-party/RedisAI/install-cpu"
+    echo "Set RedisAI CPU install path to $REDISAI_CPU_INSTALL_PATH"
```
[the remark this answers is not in the document]
Comment (mellis13) [reply setup_test_env.sh:24]: Good idea.  Done.

File: setup_test_env.sh [hunk 3]
```diff
@@ -11,3 +11,21 @@ source ./setup_env.sh
...
 export SMARTREDIS_TEST_CLUSTER=True
 echo SMARTREDIS_TEST_CLUSTER set to $SMARTREDIS_TEST_CLUSTER
+
+# Redis
+if [[ -f ./third-party/redis/src/redis-server ]]; then
+    export REDIS_INSTALL_PATH="$(pwd)/third-party/redis/src"
+    echo "Set Redis server install path to $REDIS_INSTALL_PATH"
+fi
+
+# detect RedisAI CPU installation
```
[the remark this answers is not in the document]
Comment (mellis13) [reply setup_test_env.sh:21]: Updated RedisAI version.

File: setup.py [hunk 4]
```diff
@@ -54,39 +54,34 @@ def run(self):
...
-                    os.path.join(self.build_temp, "Makefile"))
-        shutil.copytree(os.path.join(setup_path, "build-scripts"),
-                    os.path.join(self.build_temp, "build-scripts"))
+        setup_path = Path(os.path.abspath(os.path.dirname(__file__))).resolve()
 
         # build dependencies
         subprocess.check_call([f"{make_cmd} deps"],
-                              cwd=self.build_temp,
+                              cwd=setup_path,
                               shell=True)
```
[the remark this answers is not in the document]
Comment (mellis13) [reply setup.py:63]: /bin/sh seems to be the default, so should be able to use default.

**Pull request merged**